File: telerik_ui_for_winforms_courseware_chm/Courseware/TPF/VB/FisheyeBar/FisheyeBar/WelcomeScreenBottom.vb
Imports Telerik.WinControls

Public Class WelcomeScreenBottom

    Private Sub WelcomeScreenBottom_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
        Me.BackColor = Color.Transparent

        SetChildrenNotifyONMouseInput(Me.radStackItemsPanelElement2)
        SetChildrenNotifyONMouseInput(Me.radStackItemsPanelElement3)
        SetChildrenNotifyONMouseInput(Me.radStackItemsPanelElement4)
        SetChildrenNotifyONMouseInput(Me.radStackItemsPanelElement5)

        Me.radStackItemsPanelElement2.AddBehavior(New WelcomeScreenMouseOverBehavior())
        Me.radStackItemsPanelElement3.AddBehavior(New WelcomeScreenMouseOverBehavior())
        Me.radStackItemsPanelElement4.AddBehavior(New WelcomeScreenMouseOverBehavior())
        Me.radStackItemsPanelElement5.AddBehavior(New WelcomeScreenMouseOverBehavior())
    End Sub

    Private Sub SetChildrenNotifyONMouseInput(ByVal item As RadElement)
        For Each childItem As RadElement In item.Children
            childItem.NotifyParentOnMouseInput = True
            childItem.ShouldHandleMouseInput = False
            SetChildrenNotifyONMouseInput(childItem)
        Next childItem
    End Sub

    Private Sub radStackItemsPanelElement2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles radStackItemsPanelElement2.Click
        Process.Start("http://www.telerik.com/winforms")
    End Sub

    Private Sub radStackItemsPanelElement3_Click(ByVal sender As Object, ByVal e As EventArgs) Handles radStackItemsPanelElement3.Click
        Process.Start("http://www.telerik.com/products/winforms/sample-applications.aspx")
    End Sub

    Private Sub radStackItemsPanelElement4_Click(ByVal sender As Object, ByVal e As EventArgs) Handles radStackItemsPanelElement4.Click
        Process.Start("http://www.telerik.com/products/winforms/tools.aspx")
    End Sub

    Private Sub radStackItemsPanelElement5_Click(ByVal sender As Object, ByVal e As EventArgs) Handles radStackItemsPanelElement5.Click
        Process.Start("http://www.telerik.com/products/winforms/whats-new.aspx")
    End Sub

    ''' <summary>
    ''' Behavior class defining the OnMouseOver behavior of the BubbleBar
    ''' </summary>
    Public Class WelcomeScreenMouseOverBehavior
        Inherits PropertyChangeBehavior
        Public Sub New()
            MyBase.New(RadItem.IsMouseOverProperty)
        End Sub

        Private oldCursor As Cursor = Nothing

        Public Overrides Sub OnPropertyChange(ByVal element As RadElement, ByVal e As RadPropertyChangedEventArgs)
            For Each childItem As RadElement In element.Parent.Children
                Dim item As RadItem = TryCast(childItem, RadItem)
                If item IsNot Nothing Then
                    Dim animation As AnimatedPropertySetting
                    Dim scale As SizeF

                    If CBool(e.NewValue) = False Then
                        scale = New SizeF(1.0F, 1.0F)
                        If Me.oldCursor IsNot Nothing AndAlso element.ElementTree IsNot Nothing Then
                            element.ElementTree.Control.Cursor = Me.oldCursor
                        End If
                    Else
                        If item.Equals(element) Then
                            scale = New SizeF(1.1F, 1.1F)
                        Else
                            scale = New SizeF(0.9F, 0.9F)
                        End If

                        If element.ElementTree IsNot Nothing Then
                            Me.oldCursor = element.ElementTree.Control.Cursor
                            element.ElementTree.Control.Cursor = Cursors.Hand
                        End If
                    End If

                    animation = New AnimatedPropertySetting(RadElement.ScaleTransformProperty, scale, 10, 33)
                    animation.SkipToEndValueOnReplace = False
                    animation.ApplyValue(item)
                End If
            Next
        End Sub

    End Class
End Class

File: telerik_ui_for_winforms_courseware_chm/Courseware/Appearance/CS/FisheyeBar/WindowsFormsApplication1/WelcomeScreenBottom.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls;

namespace Telerik.QuickStart.WinControls
{
    public partial class WelcomeScreenBottom : UserControl
    {
        public WelcomeScreenBottom()
        {
            InitializeComponent();
        }

        private void WelcomeScreenBottom_Load(object sender, EventArgs e)
        {
            this.BackColor = Color.Transparent;

            SetChildrenNotifyONMouseInput(this.radStackItemsPanelElement2);
            SetChildrenNotifyONMouseInput(this.radStackItemsPanelElement3);
            SetChildrenNotifyONMouseInput(this.radStackItemsPanelElement4);
            SetChildrenNotifyONMouseInput(this.radStackItemsPanelElement5);

            this.radStackItemsPanelElement2.AddBehavior(new WelcomeScreenMouseOverBehavior());
            this.radStackItemsPanelElement3.AddBehavior(new WelcomeScreenMouseOverBehavior());
            this.radStackItemsPanelElement4.AddBehavior(new WelcomeScreenMouseOverBehavior());
            this.radStackItemsPanelElement5.AddBehavior(new WelcomeScreenMouseOverBehavior());
        }

        private void SetChildrenNotifyONMouseInput(RadElement item)
        {
            foreach(RadElement childItem in item.Children)
            {
                childItem.NotifyParentOnMouseInput = true;
                childItem.ShouldHandleMouseInput = false;
                SetChildrenNotifyONMouseInput(childItem);
            }
        }

        private void radStackItemsPanelElement2_Click(object sender, EventArgs e)
        {
            Process.Start("http://www.telerik.com/products/winforms.aspx");
        }

        private void radStackItemsPanelElement3_Click(object sender, EventArgs e)
        {
            Process.Start("http://www.telerik.com/products/winforms/sample-applications.aspx");
        }

        private void radStackItemsPanelElement4_Click(object sender, EventArgs e)
        {
            Process.Start("http://www.telerik.com/products/winforms/tools.aspx");
        }

        private void radStackItemsPanelElement5_Click(object sender, EventArgs e)
        {
            Process.Start("http://www.telerik.com/products/winforms/whats-new.aspx");
        }
    }

    /// <summary>
    /// Behavior class defining the OnMouseOver behavior of the BubbleBar
    /// </summary>
    public class WelcomeScreenMouseOverBehavior : PropertyChangeBehavior
    {
        public WelcomeScreenMouseOverBehavior()
            : base(RadItem.IsMouseOverProperty)
        {
        }

        private Cursor oldCursor = null;

        public override void OnPropertyChange(RadElement element, RadPropertyChangedEventArgs e)
        {
            foreach (RadElement childItem in element.Parent.Children)
            {
                RadItem item = childItem as RadItem;
                if (item != null)
                {
                    AnimatedPropertySetting animation;
                    SizeF scale;

                    if ((bool)e.NewValue == false)
                    {
                        scale = new SizeF(1F, 1F);
                        if (this.oldCursor != null && element.ElementTree != null)
                        {
                            element.ElementTree.Control.Cursor = this.oldCursor;
                        }
                    }
                    else
                    {
                        scale = item == element ? new SizeF(1.1F, 1.1F) : new SizeF(.9F, .9F);
                        if (element.ElementTree != null)
                        {
                            this.oldCursor = element.ElementTree.Control.Cursor;
                            element.ElementTree.Control.Cursor = Cursors.Hand;
                        }
                    }

                    animation = new AnimatedPropertySetting(RadElement.ScaleTransformProperty, scale, 10, 33);
                    animation.SkipToEndValueOnReplace = false;
                    animation.ApplyValue(item);
                }
            }
        }
    }
}
